tests/functional/aws-node-sdk/lib/utility/makeRequest.js

```diff
@@ -0,0 +1,87 @@
+import http from 'http';
+import https from 'https';
+import querystring from 'querystring';
+
+import conf from '../../../../../lib/Config';
+
+const transport = conf.https ? https : http;
+const ipAddress = process.env.IP ? process.env.IP : '127.0.0.1';
+
+function _parseError(responseBody) {
+    if (responseBody.indexOf('<Error>') > -1) {
+        const error = {};
+        const codeStartIndex = responseBody.indexOf('<Code>') + 6;
+        const codeEndIndex = responseBody.indexOf('</Code>');
+        error.code = responseBody.slice(codeStartIndex, codeEndIndex);
+        const msgStartIndex = responseBody.indexOf('<Message>') + 9;
+        const msgEndIndex = responseBody.indexOf('</Message>');
+        error.message = responseBody.slice(msgStartIndex, msgEndIndex);
+        return error;
+    }
+    return null;
+}
+
+/** makeRequest - utility function to generate a request
+ * @param {object} params - params for making request
+ * @param {string} params.hostname - request hostname
+ * @param {number} params.port - request port
+ * @param {string} params.method - request method
+ * @param {object} params.queryObj - query fields and their string values
+ * @param {object} params.headers - headers and their string values
+ * @param {string} params.bucket - bucket name
+ * @param {string} params.objectKey - object key name
+ * @param {function} callback - with error and response parameters
+ * @return {undefined} - and call callback
+ */
+export default function makeRequest(params, callback) {
+    const { hostname, port, method, queryObj, headers, bucket, objectKey }
+        = params;
+    const options = {
+        hostname: hostname || ipAddress, // default to where S3 Server is hosted
+        port: port || 8000, // default to port 8000
```

Comment:
OK, I hope it makes sense I added a new function `makeS3Request` [here](https://github.com/scality/S3/pull/603/commits/9b0593f9f7c541414902cf4cf11c7315a97c1f54#diff-811e22af2f7d03b99d66636c51d1ad5e) since I think we will be making the majority of the requests against S3, and I don't think it makes sense to have to specify the hostname and port for it all the time. If we need to send to another host we can still access the base `makeRequest` function and specify the hostname.